### DISCUSSION TITLE
Parent/child selector, similar to 'group'

### DISCUSSION THREAD

**Discussion by benjam-es** [tailwindlabs/tailwindcss#8970] [Ideas]
When creating a navigation structure, we can make use of `group` and `group-hover:block hidden` to make sub menus appear. To have multiple levels of sub menu, we cannot use group, and the 3rd level menu will be visible as soon as you have the first level item due to the initial `group` class.

One solution might be to have something like this:

```css
*:hover > .parent-hover\:block{
  display:block
}
```

I have no understanding of the tailwind code base in order to make a pull request, but would happy to have a go, if someone can point me to the place where the code for `group` generation is, and I could mimic that

**Comment by wongjn** [comment]
We can use arbitrary variants to achieve this – I use arbitrary variants myself to do the very UI feature you're describing! For example, `[:hover>&]:block` would be the equivalent to your example, here's a [playground](https://play.tailwindcss.com/qqIK0En9oQ) of it in action.

If you still think it's worth making the variant set to be included in Tailwind, then the code to generate the `group-` variants is here:

https://github.com/tailwindlabs/tailwindcss/blob/262079e1e5809d2a58e8d264d179c712e3d5d953/src/corePlugins.js#L145-L151
